thunder/core/transforms.py

```diff
@@ -1401,10 +1406,12 @@ def _maximum_grad(a: TensorProxy, b: TensorProxy, /):
 register_grad(pids.MAXIMUM, _maximum_grad)
 
 # This operation creates no grad associations
-register_grad(pids.ARGMAX, prims.argmax)
+register_grad(pids.ARGMAX, prims.argmax, True)
 
 # This operation creates no grad associations
-register_grad(pids.SHAPE, prims.shape)
+# NOTE we need to skip pids.SHAPE in grad transform, because `prims.shape` creates multiple outputs, which
+# breaks the assumption the in the pass that each symbol has a single output
+register_grad(pids.SHAPE, prims.shape, True)
```

Comment:
I only needed this change for `prims.shape`, but I think it could/should apply to any operation that doesn't require gradient.